web/src/vdom.ts

```diff
@@ -2,7 +2,28 @@ import _ from "lodash";
 
 import * as blueslip from "./blueslip";
 
-export function eq_array(a, b, eq) {
+export type VDomNode = {
+    key: string;
+    eq: (other: VDomNode) => boolean;
+    render: () => string;
+    conversation: Record<string, unknown>;
+};
+
+type VDomOpts = {
+    attrs: [string, string][];
+    keyed_nodes: VDomNode[];
+};
+
+export type VDOMTag = {
```

Comment:
```suggestion
export type VDomTag = {
```
nit: We should be consistent with naming I think we should either use "DOM" or "Dom".